### PR TITLE
Feat: SystemInstructions refactoring and improvements

### PR DESCRIPTION
**Pull request by mrLSD**
## Summary by CodeRabbit

* **New Features**
  * Added saturatingInt to numeric types (returns Int.max for out-of-range values).
  * Added KECCAK_EMPTY precomputed empty-hash constant.
  * Added stack consume operation for removing top elements without return.

* **Bug Fixes**
  * Improved bounds and overflow handling for copy/load/hash operations.
  * Short-circuited zero-size memory paths to skip unnecessary work.
  * Preserved stack state on early-error paths to avoid premature consumption.

* **Tests**
  * Updated and added tests covering new behaviors and edge cases.

### CODE REVIEW

File: Sources/Interpreter/MachineStack.swift [hunk 1]
```diff
@@ -182,6 +184,19 @@ struct Stack {
...
+    /// Safely consumes the specified number of elements from the top of the stack without returning them.
+    /// The removal count is clamped to the current stack size to prevent out-of-bounds runtime crashes.
+    @inline(__always)
+    mutating func consume(count: Int) {
+        assert(count >= 0 && count <= self.data.count, "Stack underflow")
+
+        let safeCount = max(0, min(count, data.count))
+        #if TRACING && TRACE_STACK_INOUT
+            self.traceStackOut.append(contentsOf: self.data.suffix(safeCount))
+        #endif
```
[the remark this answers is not in the document]
Comment (mrLSD) [reply Sources/Interpreter/MachineStack.swift:196]: Fixed: e6937e60f7dfc91cb816c4907a524056e78b8151
